{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "practical-queue",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "fatal: destination path 'Stocker' already exists and is not an empty directory.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "^C\n"
     ]
    }
   ],
   "source": [
    "!git clone https://github.com/koreal6803/Stocker.git\n",
    "!pip install -r Stocker/requirements.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "informal-april",
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "\n",
    "### 去除煩人的 warrning\n",
    "warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "comic-sight",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "date\n",
       "2000-01-04    178.0\n",
       "2000-01-05    182.0\n",
       "2000-01-06    178.0\n",
       "2000-01-07    173.0\n",
       "2000-01-10    179.0\n",
       "Name:  price, dtype: float64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "### 讀入series\n",
    "df = pd.read_csv('Stocker/price.csv', index_col='date', parse_dates=['date'])\n",
    "price = df.squeeze()\n",
    "price.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "surgical-remark",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "             price\n",
      "date              \n",
      "2000-01-04   178.0\n",
      "2000-01-05   182.0\n",
      "2000-01-06   178.0\n",
      "2000-01-07   173.0\n",
      "2000-01-10   179.0\n",
      "...            ...\n",
      "2018-08-15   241.5\n",
      "2018-08-16   239.0\n",
      "2018-08-17   239.5\n",
      "2018-08-20   239.5\n",
      "2018-08-21   241.0\n",
      "\n",
      "[4635 rows x 1 columns]\n",
      "<bound method NDFrame.squeeze of              price\n",
      "date              \n",
      "2000-01-04   178.0\n",
      "2000-01-05   182.0\n",
      "2000-01-06   178.0\n",
      "2000-01-07   173.0\n",
      "2000-01-10   179.0\n",
      "...            ...\n",
      "2018-08-15   241.5\n",
      "2018-08-16   239.0\n",
      "2018-08-17   239.5\n",
      "2018-08-20   239.5\n",
      "2018-08-21   241.0\n",
      "\n",
      "[4635 rows x 1 columns]>\n"
     ]
    }
   ],
   "source": [
    "print(df)\n",
    "print(df.squeeze)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "bearing-keeping",
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'fbprophet'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-10-cfdbbc366466>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[1;32mfrom\u001b[0m \u001b[0mStocker\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstocker\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mStocker\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      2\u001b[0m \u001b[0mtsmc\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mStocker\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mprice\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mD:\\NCKUCSIE\\grade2\\grade2-2\\DataScience\\DataScience_2021_Spring\\final_project\\Stocker\\stocker.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mpandas\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mpd\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      4\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mnumpy\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mnp\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 5\u001b[1;33m \u001b[1;32mimport\u001b[0m \u001b[0mfbprophet\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      6\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      7\u001b[0m \u001b[1;31m# matplotlib pyplot for plotting\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mModuleNotFoundError\u001b[0m: No module named 'fbprophet'"
     ]
    }
   ],
   "source": [
    "from Stocker.stocker import Stocker\n",
    "tsmc = Stocker(price)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "continuing-restoration",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'tsmc' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-11-518279a0da8d>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mmodel\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmodel_data\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mtsmc\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcreate_prophet_model\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdays\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m90\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m: name 'tsmc' is not defined"
     ]
    }
   ],
   "source": [
    "model, model_data = tsmc.create_prophet_model(days=90)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "danish-canal",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'tsmc' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-12-97f8fb70bb6f>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mtsmc\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mevaluate_prediction\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m: name 'tsmc' is not defined"
     ]
    }
   ],
   "source": [
    "tsmc.evaluate_prediction()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "steady-export",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'tsmc' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-13-3de0e70afd6f>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mtsmc\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mchangepoint_prior_analysis\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mchangepoint_priors\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0.001\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;36m0.05\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;36m0.1\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;36m0.2\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m: name 'tsmc' is not defined"
     ]
    }
   ],
   "source": [
    "tsmc.changepoint_prior_analysis(changepoint_priors=[0.001, 0.05, 0.1, 0.2])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "personal-device",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'tsmc' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-14-a5d62058da3d>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mtsmc\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpredict_future\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdays\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m100\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m: name 'tsmc' is not defined"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\F74086250\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3a9g3m_v\\\\fbprophet_9bd559ebe8c040368da9a41b4d1089f1\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\F74086250\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3a9g3m_v\\\\fbprophet_9bd559ebe8c040368da9a41b4d1089f1\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' bdist_wheel -d 'C:\\Users\\F74086250\\AppData\\Local\\Temp\\pip-wheel-f9sy4bdt'\n",
      "       cwd: C:\\Users\\F74086250\\AppData\\Local\\Temp\\pip-install-3a9g3m_v\\fbprophet_9bd559ebe8c040368da9a41b4d1089f1\\\n",
      "  Complete output (44 lines):\n",
      "  running bdist_wheel\n",
      "  running build\n",
      "  running build_py\n",
      "  creating build\n",
      "  creating build\\lib\n",
      "  creating build\\lib\\fbprophet\n",
      "  creating build\\lib\\fbprophet\\stan_model\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"C:\\Users\\F74086250\\AppData\\Local\\Temp\\pip-install-3a9g3m_v\\fbprophet_9bd559ebe8c040368da9a41b4d1089f1\\setup.py\", line 122, in <module>\n",
      "      setup(\n",
      "    File \"c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\setuptools\\__init__.py\", line 153, in setup\n",
      "      return distutils.core.setup(**attrs)\n",
      "    File \"c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\distutils\\core.py\", line 148, in setup\n",
      "      dist.run_commands()\n",
      "    File \"c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\distutils\\dist.py\", line 966, in run_commands\n",
      "      self.run_command(cmd)\n",
      "    File \"c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\distutils\\dist.py\", line 985, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\wheel\\bdist_wheel.py\", line 299, in run\n",
      "      self.run_command('build')\n",
      "    File \"c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\distutils\\cmd.py\", line 313, in run_command\n",
      "      self.distribution.run_command(command)\n",
      "    File \"c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\distutils\\dist.py\", line 985, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\distutils\\command\\build.py\", line 135, in run\n",
      "      self.run_command(cmd_name)\n",
      "    File \"c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\distutils\\cmd.py\", line 313, in run_command\n",
      "      self.distribution.run_command(command)\n",
      "    File \"c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\distutils\\dist.py\", line 985, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"C:\\Users\\F74086250\\AppData\\Local\\Temp\\pip-install-3a9g3m_v\\fbprophet_9bd559ebe8c040368da9a41b4d1089f1\\setup.py\", line 48, in run\n",
      "      build_models(target_dir)\n",
      "    File \"C:\\Users\\F74086250\\AppData\\Local\\Temp\\pip-install-3a9g3m_v\\fbprophet_9bd559ebe8c040368da9a41b4d1089f1\\setup.py\", line 36, in build_models\n",
      "      from fbprophet.models import StanBackendEnum\n",
      "    File \"C:\\Users\\F74086250\\AppData\\Local\\Temp\\pip-install-3a9g3m_v\\fbprophet_9bd559ebe8c040368da9a41b4d1089f1\\fbprophet\\__init__.py\", line 8, in <module>\n",
      "      from fbprophet.forecaster import Prophet\n",
      "    File \"C:\\Users\\F74086250\\AppData\\Local\\Temp\\pip-install-3a9g3m_v\\fbprophet_9bd559ebe8c040368da9a41b4d1089f1\\fbprophet\\forecaster.py\", line 17, in <module>\n",
      "      from fbprophet.make_holidays import get_holiday_names, make_holidays_df\n",
      "    File \"C:\\Users\\F74086250\\AppData\\Local\\Temp\\pip-install-3a9g3m_v\\fbprophet_9bd559ebe8c040368da9a41b4d1089f1\\fbprophet\\make_holidays.py\", line 14, in <module>\n",
      "      import fbprophet.hdays as hdays_part2\n",
      "    File \"C:\\Users\\F74086250\\AppData\\Local\\Temp\\pip-install-3a9g3m_v\\fbprophet_9bd559ebe8c040368da9a41b4d1089f1\\fbprophet\\hdays.py\", line 14, in <module>\n",
      "      from lunarcalendar import Lunar, Converter\n",
      "  ModuleNotFoundError: No module named 'lunarcalendar'\n",
      "  ----------------------------------------\n",
      "  ERROR: Failed building wheel for fbprophet\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\F74086250\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3a9g3m_v\\\\fbprophet_9bd559ebe8c040368da9a41b4d1089f1\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\F74086250\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3a9g3m_v\\\\fbprophet_9bd559ebe8c040368da9a41b4d1089f1\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' install --record 'C:\\Users\\F74086250\\AppData\\Local\\Temp\\pip-record-95mxn2o9\\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\Include\\fbprophet'\n",
      "         cwd: C:\\Users\\F74086250\\AppData\\Local\\Temp\\pip-install-3a9g3m_v\\fbprophet_9bd559ebe8c040368da9a41b4d1089f1\\\n",
      "    Complete output (289 lines):\n",
      "    running install\n",
      "    running build\n",
      "    running build_py\n",
      "    creating build\n",
      "    creating build\\lib\n",
      "    creating build\\lib\\fbprophet\n",
      "    creating build\\lib\\fbprophet\\stan_model\n",
      "    Importing plotly failed. Interactive plots will not work.\n",
      "    INFO:pystan:COMPILING THE C++ CODE FOR MODEL anon_model_f5236004a3fd5b8429270d00efcc0cf9 NOW.\n",
      "    WARNING:pystan:MSVC compiler is not supported\n",
      "    stanfit4anon_model_f5236004a3fd5b8429270d00efcc0cf9_4747748607347044840.cpp\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\numpy\\core\\include\\numpy\\npy_1_7_deprecated_api.h(14) : Warning Msg: Using deprecated NumPy API, disable it with #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\lib\\eigen_3.3.3\\Eigen\\src/Core/arch/CUDA/Half.h(1): warning C4819: 檔案含有無法在目前字碼頁 (950) 中表示的字元。請以 Unicode 格式儲存檔案以防止資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/meta/seq_view.hpp(145): warning C4267: 'return': 將 'size_t' 轉換為 'int'，資料可能遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/corr_matrix_constrain.hpp(1): warning C4819: 檔案含有無法在目前字碼頁 (950) 中表示的字元。請以 Unicode 格式儲存檔案以防止資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/read_corr_L.hpp(1): warning C4819: 檔案含有無法在目前字碼頁 (950) 中表示的字元。請以 Unicode 格式儲存檔案以防止資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/multiply_lower_tri_self_transpose.hpp(19): warning C4244: '正在初始化': 將 'Eigen::EigenBase<Derived>::Index' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<double,-1,-1,0,-1,-1>\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/multiply_lower_tri_self_transpose.hpp(27): warning C4244: '正在初始化': 將 'Eigen::EigenBase<Derived>::Index' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<double,-1,-1,0,-1,-1>\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/csr_u_to_z.hpp(28): warning C4267: '引數': 將 'size_t' 轉換為 'int'，資料可能遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/matrix_exp_action_handler.hpp(118): warning C4244: '正在初始化': 將 'double' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/to_matrix.hpp(136): warning C4267: '正在初始化': 將 'size_t' 轉換為 'int'，資料可能遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/to_row_vector.hpp(31): warning C4267: '正在初始化': 將 'size_t' 轉換為 'int'，資料可能遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/to_vector.hpp(31): warning C4267: '正在初始化': 將 'size_t' 轉換為 'int'，資料可能遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/welford_covar_estimator.hpp(31): warning C4244: 'return': 將 'double' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/mat/fun/welford_var_estimator.hpp(31): warning C4244: 'return': 將 'double' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/prim/arr/functor/integrate_ode_rk45.hpp(1): warning C4819: 檔案含有無法在目前字碼頁 (950) 中表示的字元。請以 Unicode 格式儲存檔案以防止資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\src\\stan/io/dump.hpp(243): warning C4101: 'exc': 未參考的區域變數\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\src\\stan/io/dump.hpp(270): warning C4101: 'exc': 未參考的區域變數\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\src\\stan/io/dump.hpp(284): warning C4101: 'exc': 未參考的區域變數\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/meta/operands_and_partials.hpp(42): warning C4267: 'return': 將 'size_t' 轉換為 'int'，資料可能遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/meta/operands_and_partials.hpp(155): warning C4267: 'return': 將 'size_t' 轉換為 'int'，資料可能遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/bessel_first_kind.hpp(19): warning C4244: '引數': 將 'double' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/bessel_first_kind.hpp(20): warning C4244: '引數': 將 'double' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/bessel_second_kind.hpp(19): warning C4244: '引數': 將 'double' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/bessel_second_kind.hpp(20): warning C4244: '引數': 將 'double' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/modified_bessel_first_kind.hpp(19): warning C4244: '引數': 將 'double' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/modified_bessel_first_kind.hpp(20): warning C4244: '引數': 將 'double' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/modified_bessel_second_kind.hpp(19): warning C4244: '引數': 將 'double' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/modified_bessel_second_kind.hpp(20): warning C4244: '引數': 將 'double' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/scal/fun/rising_factorial.hpp(18): warning C4244: '引數': 將 'double' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/cholesky_decompose.hpp(82): warning C4244: '正在初始化': 將 'Eigen::EigenBase<Derived>::Index' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<stan::math::var,-1,-1,0,-1,-1>\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/cholesky_decompose.hpp(194): warning C4244: '正在初始化': 將 'Eigen::EigenBase<Derived>::Index' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<stan::math::var,-1,-1,0,-1,-1>\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/log_softmax.hpp(104): warning C4244: '引數': 將 'Eigen::EigenBase<Derived>::Index' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<stan::math::var,-1,1,0,-1,1>\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/multiply_lower_tri_self_transpose.hpp(21): warning C4244: '正在初始化': 將 'Eigen::EigenBase<Derived>::Index' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<stan::math::var,-1,-1,0,-1,-1>\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/multiply_lower_tri_self_transpose.hpp(22): warning C4244: '正在初始化': 將 'Eigen::EigenBase<Derived>::Index' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<stan::math::var,-1,-1,0,-1,-1>\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(410): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory': 函式不使用 2 引數\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::ordered_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(246): note: 請參閱 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory' 的宣告\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::ordered_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(408): note: 編譯 類別 樣板 成員函式 'auto stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::operator ()(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)' 期間\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::ordered_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(617): note: 請參考正在進行編譯的 類別 樣板 具現化 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::ordered_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/ordered_constrain.hpp(88): note: 請參考正在進行編譯的函式 樣板 具現化 'auto stan::math::adj_jac_apply<stan::math::internal::ordered_constrain_op,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)'\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(412): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis': 函式不使用 1 引數\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::ordered_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(389): note: 請參閱 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis' 的宣告\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::ordered_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): error C2664: 'stan::math::var stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars(const double &)': 無法將引數 1 從 'Eigen::VectorXd' 轉換為 'const double &'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::ordered_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: 原因: 無法從 'Eigen::VectorXd'轉換為 'const double'\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: 沒有可以執行此轉換的使用者定義轉換運算子或無法呼叫運算子\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(333): note: 請參閱 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars' 的宣告\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::ordered_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/ordered_constrain.hpp(88): error C2440: 'return': 無法由 'void' 轉換為 'Eigen::Matrix<stan::math::var,-1,1,0,-1,1>'\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/ordered_constrain.hpp(88): note: 不能將類型 void 的運算式轉換為其他類型\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(410): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory': 函式不使用 2 引數\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::positive_ordered_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(246): note: 請參閱 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory' 的宣告\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::positive_ordered_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(408): note: 編譯 類別 樣板 成員函式 'auto stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::operator ()(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)' 期間\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::positive_ordered_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(617): note: 請參考正在進行編譯的 類別 樣板 具現化 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::positive_ordered_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/positive_ordered_constrain.hpp(86): note: 請參考正在進行編譯的函式 樣板 具現化 'auto stan::math::adj_jac_apply<stan::math::internal::positive_ordered_constrain_op,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)'\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(412): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis': 函式不使用 1 引數\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::positive_ordered_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(389): note: 請參閱 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis' 的宣告\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::positive_ordered_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): error C2664: 'stan::math::var stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars(const double &)': 無法將引數 1 從 'Eigen::VectorXd' 轉換為 'const double &'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::positive_ordered_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: 原因: 無法從 'Eigen::VectorXd'轉換為 'const double'\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: 沒有可以執行此轉換的使用者定義轉換運算子或無法呼叫運算子\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(333): note: 請參閱 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars' 的宣告\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::positive_ordered_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/positive_ordered_constrain.hpp(86): error C2440: 'return': 無法由 'void' 轉換為 'Eigen::Matrix<stan::math::var,-1,1,0,-1,1>'\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/positive_ordered_constrain.hpp(86): note: 不能將類型 void 的運算式轉換為其他類型\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(410): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory': 函式不使用 2 引數\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::simplex_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(246): note: 請參閱 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory' 的宣告\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::simplex_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(408): note: 編譯 類別 樣板 成員函式 'auto stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::operator ()(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)' 期間\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::simplex_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(617): note: 請參考正在進行編譯的 類別 樣板 具現化 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::simplex_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/simplex_constrain.hpp(97): note: 請參考正在進行編譯的函式 樣板 具現化 'auto stan::math::adj_jac_apply<stan::math::internal::simplex_constrain_op,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)'\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(412): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis': 函式不使用 1 引數\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::simplex_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(389): note: 請參閱 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis' 的宣告\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::simplex_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): error C2664: 'stan::math::var stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars(const double &)': 無法將引數 1 從 'Eigen::VectorXd' 轉換為 'const double &'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::simplex_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: 原因: 無法從 'Eigen::VectorXd'轉換為 'const double'\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: 沒有可以執行此轉換的使用者定義轉換運算子或無法呼叫運算子\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(333): note: 請參閱 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars' 的宣告\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::simplex_constrain_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/simplex_constrain.hpp(97): error C2440: 'return': 無法由 'void' 轉換為 'Eigen::Matrix<stan::math::var,-1,1,0,-1,1>'\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/simplex_constrain.hpp(97): note: 不能將類型 void 的運算式轉換為其他類型\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(410): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory': 函式不使用 2 引數\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::softmax_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(246): note: 請參閱 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::count_memory' 的宣告\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::softmax_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(408): note: 編譯 類別 樣板 成員函式 'auto stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::operator ()(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)' 期間\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::softmax_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(617): note: 請參考正在進行編譯的 類別 樣板 具現化 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::softmax_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/softmax.hpp(78): note: 請參考正在進行編譯的函式 樣板 具現化 'auto stan::math::adj_jac_apply<stan::math::internal::softmax_op,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)'\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(412): error C2660: 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis': 函式不使用 1 引數\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::softmax_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(389): note: 請參閱 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::prepare_x_vis' 的宣告\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::softmax_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): error C2664: 'stan::math::var stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars(const double &)': 無法將引數 1 從 'Eigen::VectorXd' 轉換為 'const double &'\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::softmax_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: 原因: 無法從 'Eigen::VectorXd'轉換為 'const double'\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(414): note: 沒有可以執行此轉換的使用者定義轉換運算子或無法呼叫運算子\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/functor/adj_jac_apply.hpp(333): note: 請參閱 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::build_return_varis_and_vars' 的宣告\n",
      "            with\n",
      "            [\n",
      "                F=stan::math::internal::softmax_op\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/softmax.hpp(78): error C2440: 'return': 無法由 'void' 轉換為 'Eigen::Matrix<stan::math::var,-1,1,0,-1,1>'\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\lib\\stan_math\\stan/math/rev/mat/fun/softmax.hpp(78): note: 不能將類型 void 的運算式轉換為其他類型\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\src\\stan/mcmc/sample.hpp(33): warning C4244: 'return': 將 'Eigen::EigenBase<Derived>::Index' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<double,-1,1,0,-1,1>\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\src\\stan/services/util/read_dense_inv_metric.hpp(38): warning C4267: '引數': 將 'size_t' 轉換為 'int'，資料可能遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\src\\stan/mcmc/windowed_adaptation.hpp(49): warning C4244: '=': 將 'double' 轉換為 'unsigned int'，由於類型不同，可能導致資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\src\\stan/mcmc/windowed_adaptation.hpp(50): warning C4244: '=': 將 'double' 轉換為 'unsigned int'，由於類型不同，可能導致資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/print_progress.hpp(52): warning C4244: '正在初始化': 將 'double' 轉換為 'int'，由於類型不同，可能導致資料遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/families/normal_fullrank.hpp(91): warning C4267: '正在初始化': 將 'size_t' 轉換為 'const int'，資料可能遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/families/normal_fullrank.hpp(105): warning C4244: '正在初始化': 將 'Eigen::EigenBase<Derived>::Index' 轉換為 'const int'，由於類型不同，可能導致資料遺失\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<double,-1,1,0,-1,1>\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/families/normal_fullrank.hpp(124): warning C4244: '正在初始化': 將 'Eigen::EigenBase<Derived>::Index' 轉換為 'const int'，由於類型不同，可能導致資料遺失\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<double,-1,1,0,-1,1>\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/families/normal_meanfield.hpp(48): warning C4267: '正在初始化': 將 'size_t' 轉換為 'const int'，資料可能遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/families/normal_meanfield.hpp(61): warning C4244: '正在初始化': 將 'Eigen::EigenBase<Derived>::Index' 轉換為 'const int'，由於類型不同，可能導致資料遺失\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<double,-1,1,0,-1,1>\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\src\\stan/variational/families/normal_meanfield.hpp(76): warning C4244: '正在初始化': 將 'Eigen::EigenBase<Derived>::Index' 轉換為 'const int'，由於類型不同，可能導致資料遺失\n",
      "            with\n",
      "            [\n",
      "                Derived=Eigen::Matrix<double,-1,1,0,-1,1>\n",
      "            ]\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\src\\stan/io/cmd_line.hpp(32): warning C4267: '正在初始化': 將 'size_t' 轉換為 'int'，資料可能遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\src\\stan/lang/rethrow_located.hpp(117): warning C4267: '正在初始化': 將 'size_t' 轉換為 'int'，資料可能遺失\n",
      "    c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages\\pystan\\stan\\src\\stan/model/indexing/rvalue_index_size.hpp(20): warning C4267: 'return': 將 'size_t' 轉換為 'int'，資料可能遺失\n",
      "    C:\\Users\\F74086~1\\AppData\\Local\\Temp\\pystan__ndfntnf\\stanfit4anon_model_f5236004a3fd5b8429270d00efcc0cf9_4747748607347044840.cpp(5061): warning C4267: '=': 將 'size_t' 轉換為 'int'，資料可能遺失\n",
      "    error: command 'D:\\\\NCKUCSIE\\\\grade2\\\\grade2-2\\\\Visual Studio\\\\VC\\\\Tools\\\\MSVC\\\\14.28.29910\\\\bin\\\\HostX86\\\\x64\\\\cl.exe' failed with exit code 2\n",
      "    ----------------------------------------\n",
      "ERROR: Command errored out with exit status 1: 'c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\F74086250\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3a9g3m_v\\\\fbprophet_9bd559ebe8c040368da9a41b4d1089f1\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\F74086250\\\\AppData\\\\Local\\\\Temp\\\\pip-install-3a9g3m_v\\\\fbprophet_9bd559ebe8c040368da9a41b4d1089f1\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' install --record 'C:\\Users\\F74086250\\AppData\\Local\\Temp\\pip-record-95mxn2o9\\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\Include\\fbprophet' Check the logs for full command output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting quandl\n",
      "  Using cached Quandl-3.6.1-py2.py3-none-any.whl (26 kB)\n",
      "Requirement already satisfied: matplotlib in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from -r Stocker/requirements.txt (line 2)) (3.3.4)\n",
      "Requirement already satisfied: numpy in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from -r Stocker/requirements.txt (line 3)) (1.19.5)\n",
      "Collecting fbprophet\n",
      "  Using cached fbprophet-0.7.1.tar.gz (64 kB)\n",
      "Collecting pystan\n",
      "  Using cached pystan-3.1.1-py3-none-any.whl (13 kB)\n",
      "Requirement already satisfied: pandas in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from -r Stocker/requirements.txt (line 6)) (1.1.5)\n",
      "Requirement already satisfied: pytrends in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from -r Stocker/requirements.txt (line 7)) (4.7.3)\n",
      "Requirement already satisfied: python-dateutil in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from quandl->-r Stocker/requirements.txt (line 1)) (2.8.1)\n",
      "Requirement already satisfied: requests>=2.7.0 in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from quandl->-r Stocker/requirements.txt (line 1)) (2.25.1)\n",
      "Collecting inflection>=0.3.1\n",
      "  Using cached inflection-0.5.1-py2.py3-none-any.whl (9.5 kB)\n",
      "Collecting more-itertools\n",
      "  Using cached more_itertools-8.8.0-py3-none-any.whl (48 kB)\n",
      "Requirement already satisfied: six in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from quandl->-r Stocker/requirements.txt (line 1)) (1.15.0)\n",
      "Requirement already satisfied: pytz>=2017.2 in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from pandas->-r Stocker/requirements.txt (line 6)) (2021.1)\n",
      "Requirement already satisfied: chardet<5,>=3.0.2 in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from requests>=2.7.0->quandl->-r Stocker/requirements.txt (line 1)) (4.0.0)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from requests>=2.7.0->quandl->-r Stocker/requirements.txt (line 1)) (2020.12.5)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from requests>=2.7.0->quandl->-r Stocker/requirements.txt (line 1)) (1.26.5)\n",
      "Requirement already satisfied: idna<3,>=2.5 in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from requests>=2.7.0->quandl->-r Stocker/requirements.txt (line 1)) (2.10)\n",
      "Requirement already satisfied: cycler>=0.10 in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from matplotlib->-r Stocker/requirements.txt (line 2)) (0.10.0)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from matplotlib->-r Stocker/requirements.txt (line 2)) (1.3.1)\n",
      "Requirement already satisfied: pillow>=6.2.0 in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from matplotlib->-r Stocker/requirements.txt (line 2)) (8.1.2)\n",
      "Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3 in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from matplotlib->-r Stocker/requirements.txt (line 2)) (2.4.7)\n",
      "Requirement already satisfied: Cython>=0.22 in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from fbprophet->-r Stocker/requirements.txt (line 4)) (0.29.23)\n",
      "Collecting cmdstanpy==0.9.5\n",
      "  Using cached cmdstanpy-0.9.5-py3-none-any.whl (37 kB)\n",
      "Collecting LunarCalendar>=0.0.9\n",
      "  Using cached LunarCalendar-0.0.9-py2.py3-none-any.whl (18 kB)\n",
      "Requirement already satisfied: convertdate>=2.1.2 in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from fbprophet->-r Stocker/requirements.txt (line 4)) (2.3.2)\n",
      "Collecting holidays>=0.10.2\n",
      "  Using cached holidays-0.11.1-py3-none-any.whl (133 kB)\n",
      "Requirement already satisfied: setuptools-git>=1.2 in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from fbprophet->-r Stocker/requirements.txt (line 4)) (1.2)\n",
      "Requirement already satisfied: tqdm>=4.36.1 in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from fbprophet->-r Stocker/requirements.txt (line 4)) (4.61.0)\n",
      "Collecting clikit<0.7,>=0.6\n",
      "  Downloading clikit-0.6.2-py2.py3-none-any.whl (91 kB)\n",
      "Collecting pystan\n",
      "  Using cached pystan-3.1.0-py3-none-any.whl (13 kB)\n",
      "  Using cached pystan-3.0.2-py3-none-any.whl (13 kB)\n",
      "  Using cached pystan-3.0.1-py3-none-any.whl (12 kB)\n",
      "  Using cached pystan-3.0.0-py3-none-any.whl (12 kB)\n",
      "  Using cached pystan-2.19.1.1.tar.gz (16.2 MB)\n",
      "Requirement already satisfied: pymeeus<=1,>=0.3.13 in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from convertdate>=2.1.2->fbprophet->-r Stocker/requirements.txt (line 4)) (0.5.11)\n",
      "Requirement already satisfied: korean-lunar-calendar in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from holidays>=0.10.2->fbprophet->-r Stocker/requirements.txt (line 4)) (0.2.1)\n",
      "Requirement already satisfied: hijri-converter in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from holidays>=0.10.2->fbprophet->-r Stocker/requirements.txt (line 4)) (2.1.2)\n",
      "Requirement already satisfied: ephem>=3.7.5.3 in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from LunarCalendar>=0.0.9->fbprophet->-r Stocker/requirements.txt (line 4)) (3.7.7.1)\n",
      "Requirement already satisfied: lxml in c:\\users\\f74086250\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from pytrends->-r Stocker/requirements.txt (line 7)) (4.6.3)\n",
      "Building wheels for collected packages: fbprophet, pystan\n",
      "  Building wheel for fbprophet (setup.py): started\n",
      "  Building wheel for fbprophet (setup.py): finished with status 'error'\n",
      "  Running setup.py clean for fbprophet\n",
      "  Building wheel for pystan (setup.py): started\n",
      "  Building wheel for pystan (setup.py): still running...\n",
      "  Building wheel for pystan (setup.py): still running...\n",
      "  Building wheel for pystan (setup.py): still running...\n",
      "  Building wheel for pystan (setup.py): still running...\n",
      "  Building wheel for pystan (setup.py): still running...\n",
      "  Building wheel for pystan (setup.py): still running...\n",
      "  Building wheel for pystan (setup.py): still running...\n",
      "  Building wheel for pystan (setup.py): still running...\n",
      "  Building wheel for pystan (setup.py): still running...\n",
      "  Building wheel for pystan (setup.py): still running...\n",
      "  Building wheel for pystan (setup.py): still running...\n",
      "  Building wheel for pystan (setup.py): finished with status 'done'\n",
      "  Created wheel for pystan: filename=pystan-2.19.1.1-cp39-cp39-win_amd64.whl size=28883384 sha256=05d191ca5baca1369e5c2b889d18cee96e9038bb99b57d85e1e6894a0a4a93d9\n",
      "  Stored in directory: c:\\users\\f74086250\\appdata\\local\\pip\\cache\\wheels\\b8\\36\\bf\\7ec7e363f796373cea3eb9ea94e83f5bbbb586d2edbf7e3417\n",
      "Successfully built pystan\n",
      "Failed to build fbprophet\n",
      "Installing collected packages: pystan, more-itertools, LunarCalendar, inflection, holidays, cmdstanpy, quandl, fbprophet\n",
      "    Running setup.py install for fbprophet: started\n",
      "    Running setup.py install for fbprophet: finished with status 'error'\n"
     ]
    }
   ],
   "source": [
    "tsmc.predict_future(days=100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "deadly-rates",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
